relay-config/src/config.rs

```diff
@@ -966,8 +966,27 @@ pub struct EnvelopeSpool {
     ///
     /// This value should be lower than [`Health::max_memory_percent`] to prevent flip-flopping.
     ///
-    /// Warning: this threshold can cause the buffer service to deadlock when the buffer itself
-    /// is using too much memory (influenced by [`Self::batch_size_bytes`]).
+    /// Warning: This threshold can cause the buffer service to deadlock when the buffer consumes
+    /// excessive memory (as influenced by [`Self::batch_size_bytes`]).
+    ///
+    /// This scenario arises when the buffer stops spooling due to reaching the
+    /// [`Self::max_backpressure_memory_percent`] limit, but the batch threshold for spooling
+    /// ([`Self::batch_size_bytes`]) is never reached. As a result, no data is spooled, memory usage
```

Comment:
Can we add a validation to the config here to make this impossible (-> error)?

Reply:
We can definitely do this, I have to see about the best way.